corehq/apps/data_interfaces/tasks.py

```diff
@@ -167,11 +167,16 @@ def run_case_update_rules_for_domain(domain, now=None):
     serializer='pickle',
 )
 def run_case_update_rules_for_domain_and_db(domain, now, run_id, case_type, db=None):
-    all_rules = AutomaticUpdateRule.by_domain(domain, AutomaticUpdateRule.WORKFLOW_CASE_UPDATE)
-    rules = list(all_rules.filter(case_type=case_type))
+    rules = list(
+        AutomaticUpdateRule.by_domain(domain, AutomaticUpdateRule.WORKFLOW_CASE_UPDATE).filter(case_type=case_type)
+    )
+
+    modified_before = AutomaticUpdateRule.get_boundary_date(rules, now)
+    modified_after = AutomaticUpdateRule.get_oldest_rule_run(rules)
```

Comment:
It looks like this gets the latest run for each rule and then takes the earliest of them. So conceptually (to simplify, let's say there's just one rule) this is having us only check cases modified since the last time the rule was run.

I don't think that that's logically airtight though, because the rules can be based not only on the contents of the case, but the value of when "now" is. So for example, a rule might trigger on all cases with a last_visit date (a case property) more than 90 days ago. Yesterday when the rule was run, it might have been less than 90 days ago, but now today it's more than 90 days.

Does that make sense?

Reply:
Offlined with Danny, but just commenting that this makes lots of sense and was a severe oversight on my part. I'm going to close this PR out for now.